server/src/main/java/org/elasticsearch/action/admin/cluster/settings/ClusterUpdateSettingsRequest.java

```diff
@@ -57,6 +62,27 @@ public ActionRequestValidationException validate() {
         return validationException;
     }
 
+    /**
+     * Sets the value of "flat_settings".
+     * 
+     * @param flatSettings
+     *            value of "flat_settings" flag to be set
+     * @return this request
+     */
+    public ClusterUpdateSettingsRequest flatSettings(boolean flatSettings) {
+        this.flatSettings = flatSettings;
+        return this;
+    }
+
+    /**
+     * Return settings in flat format.
+     * 
+     * @return <code>true</code> if settings need to be returned in flat format; <code>false</code> otherwise.
+     */
```

Comment:
can you clarify that this is only used by the high-level REST client? Maybe we should also do the same in GetIndexRequest (`flatSettings` and `includeDefaults`), I thought we did but we didn't.